File: app/src/test/java/com/itmoldova/utils/UtilsTest.java
package com.itmoldova.utils;

import com.itmoldova.TestUtils;
import com.itmoldova.model.Item;
import com.itmoldova.util.Utils;

import org.junit.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class UtilsTest {

    public static final int N = 5;

    @Test
    public void testGetRelatedArticles_WhenCurrentItemInTheList() {
        Item currentItem = TestUtils.oneArticleWithImage().get(0);
        List<Item> items = TestUtils.tenArticles();

        // current item is in the list -> should remove it and return the first six items

        List<Item> related = Utils.INSTANCE.getRelatedArticles(items, currentItem, N);
        assertThat(related.size()).isEqualTo(N - 1);
        assertThat(related).doesNotContain(currentItem);
    }

    @Test
    public void testGetRelatedArticles_WhenCurrentItemNotInTheList() {
        Item currentItem = new Item();
        currentItem.setGuid("unique-100");
        List<Item> items = TestUtils.tenArticles();

        // current item is not in the list -> should return the first six items

        List<Item> related = Utils.INSTANCE.getRelatedArticles(items, currentItem, N);
        assertThat(related.size()).isEqualTo(N - 1);
        assertThat(related).doesNotContain(currentItem);
    }

    @Test
    public void testGetRelatedArticles_InvalidParams() {
        List<Item> related = Utils.INSTANCE.getRelatedArticles(null, null, N);
        assertThat(related.size()).isEqualTo(0);
    }
}

File: app/src/test/java/com/itmoldova/utils/UtilsTest.kt
package com.itmoldova.utils

import com.itmoldova.TestUtils
import com.itmoldova.model.Article
import com.itmoldova.util.Utils
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class UtilsTest {

    @Test
    fun testGetRelatedArticles_WhenCurrentArticleInTheList() {
        val currentArticle = TestUtils.oneArticleWithImage()[0]
        val articles = TestUtils.tenArticles()

        // current article is in the list -> should remove it and return the first six articles

        val related = Utils.getRelatedArticles(articles, currentArticle, N)
        assertThat(related.size).isEqualTo(N - 1)
        assertThat(related).doesNotContain(currentArticle)
    }

    @Test
    fun testGetRelatedArticles_WhenCurrentArticleNotInTheList() {
        val currentArticle = Article()
        currentArticle.guid = "unique-100"
        val articles = TestUtils.tenArticles()

        // current article is not in the list -> should return the first six articles

        val related = Utils.getRelatedArticles(articles, currentArticle, N)
        assertThat(related.size).isEqualTo(N - 1)
        assertThat(related).doesNotContain(currentArticle)
    }

    @Test
    fun testGetRelatedArticles_InvalidParams() {
        val related = Utils.getRelatedArticles(null, null, N)
        assertThat(related.size).isEqualTo(0)
    }

    companion object {
        val N = 5L
    }
}



